webview-ui/src/ClusterProperties/AgentPoolDisplay.tsx

```diff
@@ -1,22 +1,51 @@
 import styles from "./ClusterProperties.module.css";
-import { AgentPoolProfileInfo } from "../../../src/webview-contract/webviewDefinitions/clusterProperties";
+import { AgentPoolProfileInfo, ClusterInfo } from "../../../src/webview-contract/webviewDefinitions/clusterProperties";
+import { VSCodeButton } from "@vscode/webview-ui-toolkit/react";
+import { EventDef, vscode } from "./state";
+import { EventHandlers } from "../utilities/state";
 
 export interface AgentPoolDisplayProps {
-    profileInfo: AgentPoolProfileInfo
+    clusterInfo: ClusterInfo;
+    profileInfo: AgentPoolProfileInfo;
+    eventHandlers: EventHandlers<EventDef>;
+    clusterOperationRequested: boolean;
 }
 
+// Abort on anything except: Canceled, Failed, Succeeded
+const terminalProvisioningStates = ["Canceled", "Failed", "Succeeded"];
+
 export function AgentPoolDisplay(props: AgentPoolDisplayProps) {
+    const isProvisioningStateFromCluster = props.clusterInfo.provisioningState === props.profileInfo.provisioningState;
+    const isOperationInProgress = !terminalProvisioningStates.includes(props.profileInfo.provisioningState);
+    const showAbortButton = !isProvisioningStateFromCluster && isOperationInProgress;
+
+    function handleAbortClick(agentPoolName: string) {
+        vscode.postAbortAgentPoolOperation(agentPoolName);
+        props.eventHandlers.onSetClusterOperationRequested();
+    }
+
     return (
         <dl className={styles.propertyList}>
+            <dt>Provisioning State</dt>
+            <dd>
+                {props.profileInfo.provisioningState}
+                {showAbortButton &&
+                <>
+                    &nbsp;
+                    <VSCodeButton disabled={props.clusterOperationRequested} onClick={() => handleAbortClick(props.profileInfo.name)}>Abort</VSCodeButton>
```

Comment:
Sorry, I completely forgot about this, but I think this button should have the attribute `appearance="secondary"`. That'll change the colour from that really bright blue to something a bit more subtle.

While we're at it, I think the same attribute should be applied to the Start and Stop buttons (in `ClusterDisplay.tsx`).

Reply:
The `Secondary` gives this and primary gives this:

I almost feel `Primary` is better :) see how abort looks almost camouflaged, but I am not good in colour choice, let me know and I can make primary or secondary.

![image](https://github.com/Azure/vscode-aks-tools/assets/6233295/a0afafe6-30c5-4e4f-8410-61153a45bffc)

Reply:
sounds good, lets go with `primary` made the change and pushed.